pkg/cfg/common.go

```diff
@@ -177,9 +177,10 @@ type Common struct {
 	KeepAliveInterval         time.Duration `yaml:"keepAliveInterval"`
 	MaxIdleConnsPerHost       int           `yaml:"maxIdleConnsPerHost"`
 
-	ExpireDelaySec             int32 `yaml:"expireDelaySec"`
-	InternalRoutingCache       int32 `yaml:"internalRoutingCache"`
-	GraphiteWeb09Compatibility bool  `yaml:"graphite09compat"`
+	ExpireDelaySec             int32    `yaml:"expireDelaySec"`
+	InternalRoutingCache       int32    `yaml:"internalRoutingCache"`
+	TLDCacheExtraPrefixes      []string `yaml:"tldCacheExtraPrefixes"`
```

Comment:
If I understand correctly, we can add prefixes that have more than one level of hierarchy. For example, we can have `a.b.c`, `a.b.d`, and `hello` here.

We may consider using a tree of strings here instead of a list. So, we will go from:

```
tldCacheExtraPrefixes:
  - "a.b.c"
  - "a.b.d"
  - "hello"
```
to for example
```
tldCacheExtraPrefixes:
  - "a":
      "b":
        - "c"
        - "d"
  - "hello"
```
(of course, there are also other ways to specify hierarchy, like with a nested list).

This will be more type-safe. It would make parsing a bit harder but then the other parsing and sorting logic will become much simpler.

At the same time, we do not expect many levels of hierarchy. So, it will still be convenient when we want to have e.g. a single prefix `a.b.c`.

Reply:
Yes it can be implemented like this. however, I don't see a huge benefit from implementing it this way.
Also, it'll create and ambiguity that when we configure:
```tldCacheExtraPrefixes:
  - "a":
      "b":
        - "c"
        - "d"
  - "hello"```
will it induce probing of `a` or `a.b` or is it only `a.b.c` and `a.b.d`.
Although it might not be a normal use-case, probing `a.b.c` should not necessarily cause probing of `a.b`. Of course we can add some attributes to each prefix to identify if it should be probed, but I believe that is not neat.

Reply:
@emadolsky Yes, there are pros and cons to both. I'm totally ok if we leave this like it is.